org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/util/gradle/PublishedGradleVersionsWrapper.java

```diff
@@ -70,6 +71,16 @@ protected void canceling() {
             Thread.currentThread().interrupt();
         }
 
+        private LookupStrategy getLookupStrategy() {
+            String strategyName = InstanceScope.INSTANCE.getNode(CorePlugin.PLUGIN_ID).get("publishedgradleversions.lookupstrategy", LookupStrategy.REMOTE_IF_NOT_CACHED.name());
```

Comment:
So to avoid remote interaction, I'll need to set `publishedgradleversions.lookupstrategy=CACHED` somewhere, correct?

Reply:
That's the idea, yes. Right now there's no UI, but you can use tools like oomph to configure IDE preferences. Without that, you'll need to figure out in which folder does Eclipse (and VSCode Java for that matter) puts the workspace. you need to add the `org.eclipse.buildship.core.publishedgradleversions.lookupstrategy=CACHED_ONLY` entry to the `${workspace_loc}/.metadata/.plugins/org.eclipse.core.runtime/.settings/org.eclipse.buildship.core.prefs` file.